quarkchain/evm/specials.py

```diff
@@ -254,6 +256,27 @@ def proc_transfer_mnt(ext, msg):
     return apply_msg(ext, new_msg)
 
 
+def proc_deploy_root_chain_staking_contract(ext, msg):
+    from quarkchain.evm.messages import create_contract
+
+    gascost = 3
+    if msg.gas < gascost:
+        return 0, 0, []
+    new_msg = vm.Message(
+        msg.to,  # current special address
+        b"",
+        0,
+        msg.gas - gascost,
+        ROOT_CHAIN_POSW_CONTRACT_BYTECODE,
+        msg.depth + 1,
+        to_full_shard_key=msg.to_full_shard_key,
+        transfer_token_id=msg.transfer_token_id,
+        gas_token_id=msg.gas_token_id,
+    )
+    # CREATE2, deterministic staking contract address
+    return create_contract(ext, new_msg, b"", salt=bytes(32))
```

Comment:
Why not create a contract with a specific address like precompiled contracts?

Reply:
then the handling logic of the tx targeting at precompiled address will be kinda complicated, something like this:

```
# messages.py

    if msg.code_address in ext.specials:
        res, gas, dat = ext.specials[msg.code_address](ext, msg)
    else:
        res, gas, dat = vm.vm_execute(ext, msg, code)

# specials.py

def proc(ext, msg):
  if ext.has_code(msg.to):
    return vm.vm_execute(ext, msg, code)
  else:
    # make a deployment message
```

which complicates the logic a little bit and I don't see much benefit here, except that the address is slightly easier to remember